### ISSUE TITLE
Custom sorting of groups for basic data operations

### ISSUE THREAD

**Issue by ogrinishin** [telerik/nativescript-ui-feedback#522]
### Did you verify this is a real problem by searching [Stack Overflow](http://stackoverflow.com/questions/tagged/nativescript)?
Yes

### Tell us about the problem
I need to change sorting of groups that are rendering by `groupingFunction`, because they are sorted by alphabet, and `sortingFunction` method sort data inside each group

**Comment by alexandruantonica** [comment]
Is there any change to be this feature implemented?

**Comment by glennsyang** [comment]
Yes this is an important feature for me too. Right now, it only sorts the groupings alphabetically, but I need to sort by some other data as well.
Any news on this?

**Comment by nmongiya** [comment]
+1 I am also looking forward for this important feature.

**Comment by bonsaibox** [comment]
also would like to have this feature

**Comment by jpierront** [comment]
You can do some magic to order your group ;-)

Add an "order" prefix to your group label, and remove it before displaying it.

this.groupingFunction = (item: ItemData) => {
  return item.group.order + '@' + item.group.label;
};

<ng-template tkGroupTemplate let-category="category">
  <StackLayout>
    <Label [text]="category.split('@').splice(1).join('@')"></Label>
  </StackLayout>
</ng-template>

Be careful, it's an alphabetical sort so ... '9' is greater than '80' ;-)

**Comment by sflament** [comment]
+1
I am also looking forward for this important feature, with Jonas Enhancement
I need to push a structured data in the groupTemplate

**Comment by M-kothakapu** [comment]
Does anyone know if this feature is implemented?? Even I'm looking for a similar grouping of list data items for my project.

**Comment by Pandishpan** [comment]
Any updates on this feature?

**Comment by n0mer** [comment]
```
  static _MS_PER_DAY = 1000 * 60 * 60 * 24;
  static maxDay = new Date(8640000000000000);
  static utc2 = Date.UTC(BrowseComponent.maxDay.getFullYear(), BrowseComponent.maxDay.getMonth(), BrowseComponent.maxDay.getDate());

  public static dateDiffInDays(a: Date): number {

    // Discard the time and time-zone information.
    const utc1 = Date.UTC(a.getFullYear(), a.getMonth(), a.getDate());

    return Math.floor((BrowseComponent.utc2 - utc1) / BrowseComponent._MS_PER_DAY);
  }

  public dateGroupingFunc(bravo: Bravo): string {
    let date = new Date(bravo.created_at);
     // https://angular.io/api/common/DatePipe
    const prefix = `d${BrowseComponent.dateDiffInDays(date)}`;
    console.log(prefix);
    return `${prefix}@${formatDate(date, 'fullDate', 'en-US')}`;
  }

```
my work-around for dates grouping

**Comment by coffeeneed** [comment]
I really need this feature too! Any news?

**Comment by Pandishpan** [comment]
```
getSortingString(title: string): string {
	let data = "";
	for (let i = 0; i < title.length; i++) {
		data += String.fromCharCode(90 - parseInt(title.charAt(i), 10));
	}

	return data;
}


this.myGroupingFunc = (item) => {
		const dt = new Date(item.created_at);
		return this.getSortingString(formatDate(dt, "yyyyMMdd", this.locale)) + "@" + formatDate(dt, "dd-MM-yyyy", this.locale);
};

```
My workaround for sorting inside the group if you want to sort them by newest on top.
Using a prefix (thanks @jpierront ) generated based on year, month and day of the item. The prefix is converted to a string.

**Comment by alexhribsek** [comment]
Thank you very much @jpierront and @Pandishpan ❤️

**Comment by diegovincent** [comment]
So. It will not be implemented. Could you please simply say that instead of leaving the community hanging? Or maybe you could say you accept PRs, and the community can help.

**Comment by akhtariali** [comment]
> ```
> getSortingString(title: string): string {
> 	let data = "";
> 	for (let i = 0; i < title.length; i++) {
> 		data += String.fromCharCode(90 - parseInt(title.charAt(i), 10));
> 	}
> 
> 	return data;
> }
> 
> 
> this.myGroupingFunc = (item) => {
> 		const dt = new Date(item.created_at);
> 		return this.getSortingString(formatDate(dt, "yyyyMMdd", this.locale)) + "@" + formatDate(dt, "dd-MM-yyyy", this.locale);
> };
> ```
> 
> My workaround for sorting inside the group if you want to sort them by newest on top.
> Using a prefix (thanks @jpierront ) generated based on year, month and day of the item. The prefix is converted to a string.

@Pandishpan Could you please explain the logic behind the code?
